nx/lib/nx/binary_backend.ex

```diff
@@ -1671,12 +1671,40 @@ defmodule Nx.BinaryBackend do
   end
 
   @impl true
-  def indexed_add(
-        %T{} = out,
-        %T{shape: shape, type: {_, target_size}} = target,
-        %T{shape: {indices_rows, _indices_cols} = indices_shape} = indices,
-        %T{shape: {indices_rows}} = updates
-      ) do
+  def indexed_add(out, target, indices, updates) do
+    resolve_updates = fn upds -> Enum.reduce(upds, 0, &+/2) end
+    update_element = fn cur, upd -> cur + upd end
+
+    indexed_op(out, target, indices, updates, resolve_updates, update_element)
+  end
+
+  @impl true
+  def indexed_put(out, target, indices, updates) do
+    resolve_updates = fn upds -> Enum.at(upds, -1) end
+    update_element = fn _cur, upd -> upd end
+
+    indexed_op(out, target, indices, updates, resolve_updates, update_element)
+  end
+
+  # Generalized `indexed_*` operation
+
+  # # Parameters
+
+  # - `out` - output tensor
+  # - `target` - target tensor
+  # - `indices` - tensor with indices to update
+  # - `updates` - tensor with updates to make
+  # - `resolve_updates` - anonymous function to resolve repeated updates
+  # - `update_element` - anonymous function that updates an element, given
+  #    the current element and the update
```

Comment:
I think it's fine to skip this, since it's a private function and from context it's pretty straightforward what the arguments are :)